app/Http/Controllers/AlbumsController.php

```diff
@@ -57,4 +59,90 @@ public function get()
 
 		return $return;
 	}
+
+	/**
+	 * @return array|string returns an array of photos of all albums or false on failure
+	 */
+	public function getPositionData()
+	{
+		// caching to avoid further request
+		Configs::get();
+
+		// Initialize return var
+		$return = array();
+		$return['albums'] = array();
+
+		// Get photos
+		// Get album information
+		$UserId = $this->sessionFunctions->id();
+		$full_photo = Configs::get_value('full_photo', '1') == '1';
+
+		$toplevel = $this->albumFunctions->getToplevelAlbums();
+		if ($toplevel === null) {
+			return Response::error('I could not find you.');
+		}
+
+		// All normal albums
+		$toplevel_albums = $this->albumFunctions->prepare_albums($toplevel['albums']);
+		// Add albums shared with users
+		$toplevel_albums = array_merge($toplevel_albums, $this->albumFunctions->prepare_albums($toplevel['shared_albums']));
+
+		$toplevel_album_ids = array();
+		$album_list = array();
+
+		foreach ($toplevel_albums as $album_iterator) {
```

Comment:
What is it that you are trying to do here? Generate an array of all accessible albums? `AlbumFunctions::getPublicAlbums()` already does that, and much more efficiently than the code here.